File: clients/backend/ingestMessage.ts
import { MessageAnalysis } from "../perspectiveAPI.js";

export default async function ingestMessage(message: MessageAnalysis) {
    try {
        const response = await fetch(`${process.env.BACKEND_URL!}/psychopass/message`, {
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                "User-Agent": `${process.env.npm_package_name}/${process.env.npm_package_version!} node.js/${process.version}`,
                "Authorization": `Token ${process.env.BACKEND_API_KEY!}`
            },
            body: JSON.stringify(message)
        });
        if (!response.ok) throw new Error(`Ingest Message: ${response.status} ${response.statusText}`);
    } catch (error) {
        console.error(error);
    }
}

File: clients/backend/ingest_message.py
from os import getenv
from sys import version
from requests import post


def ingest_message(message: dict) -> None:

    response = post(
        f"{getenv('BACKEND_URL')}/psychopass/message",
        headers={
            "Accept": "application/json",
            "User-Agent": f"sibyl-discord python/{version}",
            "Authorization": f"Token {getenv('BACKEND_API_KEY')}",
        },
        json=message,
    )

    response.raise_for_status()
